src/redux/Books/books.js

```diff
@@ -1,36 +1,84 @@
+const URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/OyUiXqyBVJcIT8DxplTW/books/';
+
 const ADD = 'redux/Books/book/ADD';
 const REMOVE = 'redux/Books/book/REMOVE';
+const GET = 'redux/Books/book/GET';
 const initialState = [];
-
 export default function bookReducer(state = initialState, action) {
   switch (action.type) {
     case 'redux/Books/book/ADD':
       return [
         ...state,
-        {
-          id: action.id,
-          title: action.payload.title,
-          author: action.payload.author,
-        },
+        action.payload,
       ];
 
     case 'redux/Books/book/REMOVE':
-      return state.filter((book) => book.id !== action.id);
+      return state.filter((book) => book.id !== action.payload);
+
+    case 'redux/Books/book/GET':
+      return action.payload;
 
     default: return state;
   }
 }
 
-const addBook = (id, payload) => ({
-  type: ADD,
-  payload,
-  id,
-});
+const addBook = (book) => (dispatch) => {
+  fetch(URL, {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      item_id: book.id,
+      title: book.title,
+      author: book.author,
+      category: 'Fiction',
+    }),
+  }).then(() => {
```

Comment:
* As stated in the [project requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/connec_to_api.md#:~:text=Use%20fetch%20or%20axios%20with%20createAsyncThunk%20for%20making%20API%20calls.) Kindly use createAsyncThunk for your API calls

Reply:
@Davidosky007,
 i think it's only required for fetching the data and not posting it